### ISSUE TITLE
top(none)/detached/window(chrome://browser/content/browser.xul)/dom/event-listeners lingers until restart

### ISSUE THREAD

**Issue by Fanolian** [piroor/treestyletab#886]
The test is done with about:memory in Nightly and TST nightly version (currently 0.15.2015031900a181223)

Steps to reproduce:
1) In a new profile, install TST in Nightly. Leave all settings as default
2) Open one tab and only in one window. Go to about:memory. 
3) Minimize memory usage a few times then measure. The result at Main Process - Other Measurements - event-counts should be similar to :
```
1,551 (100.0%) -- event-counts
└──1,551 (100.0%) -- window-objects
   ├──1,299 (83.75%) -- top(chrome://browser/content/browser.xul, id=3)/active
   │  ├──1,298 (83.69%) -- window(chrome://browser/content/browser.xul)/dom
   │  │  ├──1,295 (83.49%) ── event-listeners
   │  │  └──────3 (00.19%) ── event-targets
   │  └──────1 (00.06%) ── window(about:blank)/dom/event-targets
   ├────214 (13.80%) -- top(about:newtab, id=17)/active/window(about:newtab)/dom
   │    ├──212 (13.67%) ── event-listeners
   │    └────2 (00.13%) ── event-targets
   ├─────21 (01.35%) ++ (4 tiny)
   └─────17 (01.10%) ++ top(about:memory, id=9)/active/window(about:memory)/dom
```
4) Open a new window or a new private window, then close it.
5) Minimize memory usage and measure again. a `top(none)/detached/window(chrome://browser/content/browser.xul)/dom/event-listeners` is created.
6) Repeat step 4 and 5 a few times. More `top(none)/detached/…` are created and cannot be destroyed unless restarting Nightly. This may end up like:
```
9,982 (100.0%) -- event-counts
└──9,982 (100.0%) -- window-objects
   ├──8,632 (86.48%) ── top(none)/detached/window(chrome://browser/content/browser.xul)/dom/event-listeners [12]
   ├──1,299 (13.01%) -- top(chrome://browser/content/browser.xul, id=3)/active
   │  ├──1,298 (13.00%) -- window(chrome://browser/content/browser.xul)/dom
   │  │  ├──1,295 (12.97%) ── event-listeners
   │  │  └──────3 (00.03%) ── event-targets
   │  └──────1 (00.01%) ── window(about:blank)/dom/event-targets
   └─────51 (00.51%) ++ (6 tiny)
```

Without TST, `top(none)/detached/…` is gone after minimizing memory usage. 

I don't know if this issue impacts TST's functions or Firefox's performance (as I know nothing about coding), but this is definitely something abnormal.

**Comment by Fanolian** [comment]
Is it the same or similar to [Bug 1142597 - sdk/tabs memory leak when opening and closing new windows](https://bugzilla.mozilla.org/show_bug.cgi?id=1142597)?

**Comment by jryans** [comment]
> Is it the same or similar to Bug 1142597 - sdk/tabs memory leak when opening and closing new windows?

I don't think it's this bug because TST does not make use of the SDK. But, it could be something similar.

**Comment by Fanolian** [comment]
I don't know if commit 94428a1 is related, but I can still reproduce the issue on TST nightly 0.15.2015061300a003855

**Comment by piroor** [comment]
Confirmed on Nightly 45.0a1 with TST 9f14f59. But currently I have not enough time to research the reason why TST grabs the memory...

**Comment by Fanolian** [comment]
This was fixed in 0.16.2015111001 already. Thanks!
